src/coffeescript.coffee

```diff
@@ -113,6 +113,7 @@ exports.compile = compile = withPrettyErrors (code, options = {}) ->
   # which might’ve gotten misaligned from the original source due to the
   # `clean` function in the lexer).
   if options.ast
+    nodes.allCommentTokens = helpers.extractAllCommentTokens tokens
```

Comment:
I looked at how we might be able to pass the comment tokens directly within `grammar.coffee` - there's probably a convoluted way to do it involving eg adding a new option to the `o()` helper and only using that helper in the `Root` rules, but this is far simpler to just pass the comments at this "higher level" where we have visibility of both the `tokens` and the resulting `nodes`

I'm not sure whether it's a problem for the `nodes` tree not to be getting generated "purely" inside `grammar.coffee` - I guess the use case where this would be problematic is if someone wanted to manually get a handle on a nodes tree (eg via `CoffeeScript.nodes()` and then call `nodes.ast()` themselves?

If that is a use case that needs to be supported, I think the cleanup of the AST root's location data just below this would already be violating that (since it's creating a difference between what you get if you call `CoffeeScript.nodes()` -> `nodes.ast()` vs just `CoffeeScript.ast()`)